completion_aggregator/settings/common.py

```diff
@@ -32,6 +32,30 @@ def plugin_settings(settings):
         'sequential',
         'vertical',
     }
+
+    # Whitelist the allowed xAPI-transformed events for use with the xAPI backends
+    enabled_aggregator_events = [
+        f'openedx.completion_aggregator.{event_type}.{block_type}'
+
+        for event_type in settings.ALLOWED_COMPLETION_AGGREGATOR_EVENT_TYPES
+        for block_type in settings.ALLOWED_COMPLETION_AGGREGATOR_EVENT_TYPES[event_type]
+    ]
+    global_processors = settings.EVENT_TRACKING_BACKENDS.get(
+        'event_transformer', {}).get(
+            'OPTIONS', {}).get(
+                'processors', [])
+    xapi_processors = settings.EVENT_TRACKING_BACKENDS.get(
+        'event_transformer', {}).get(
+            'OPTIONS', {}).get(
+                'backends', {}).get(
+                    'xapi', {}).get(
+                        'OPTIONS', {}).get(
+                            'processors', [])
+    for processor in global_processors + xapi_processors:
+        if 'whitelist' in processor.get('OPTIONS', {}):
+            for event_name in enabled_aggregator_events:
+                processor['OPTIONS']['whitelist'].append(event_name)
+
```

Comment:
@bmtcril @Ian2012 I'm trying to add these completion_aggregator events to the whitelists, but a) it isn't working, and b) this is really ugly.

So I'm thinking about creating a function in event-routing-backends for this purpose.

1. Do you have any opinions for how this _should_ work / look? 
1. Is there anything else that should be refreshed/changed when this setting gets updated?

I'll have another bash at this tomorrow.

Reply:
FYI, I've moved this logic into here, but it's still not always working: https://github.com/openedx/event-routing-backends/pull/431/commits/a415ee72083c20961625b14072a4061931686f11

I think it's breaking because the django app plugins aren't loaded in a consistent order -- this approach requires the ERB app to be loaded first, then the completion aggregator, and I don't think we can rely on that.

Reply:
Hmm. The apps should be loaded in the order of settings.INSTALLED_APPS, so traditionally you would be able to ensure a dependent app loads correctly by putting it at the bottom of the list. I think that when dealing with Tutor plugins they run in the order of the plugins, so when a Tutor plugin like Aspects is shoving things into INSTALLED_APPS it should be in an idempotent order based on where the Tutor plugin falls.

I would expect ERB to be ready first if the Aspects plugin is higher in the Tutor plugins order than whatever is injecting completion-aggregator into the settings (if being installed via plugin).

Reply:
> I would expect ERB to be ready first if the Aspects plugin is higher in the Tutor plugins order than whatever is injecting completion-aggregator into the settings (if being installed via plugin).

Yep, I hope so too.. I'm rebuilding my stack now, and we'll see.

> so when a Tutor plugin like Aspects is shoving things into INSTALLED_APPS it should be in an idempotent order based on where the Tutor plugin falls.

Does Aspects shove things into INSTALLED_APPS?  I thought ERB was just a django plugin like this repo, making it just another pip requirement, so who knows what order they're loaded in?

But anyway, these are empirical questions that can be answered by doing :) Hoping I'll have a happy stack today and can test these assumptions.

Reply:
FYI I rebuilt my tutor dev nightly stack and these changes worked a treat :)